server/core/basic_cluster.go

```diff
@@ -412,58 +463,68 @@ func (bc *BasicCluster) CheckAndPutRegion(region *RegionInfo) []*RegionInfo {
 	return bc.PutRegion(region)
 }
 
+// AtomicCheckAndPutRegion checks if the region is valid to put, if valid then put.
+func (bc *BasicCluster) AtomicCheckAndPutRegion(region *RegionInfo) ([]*RegionInfo, error) {
```

Comment:
what's the difference with PreCheckPutRegion？ only setRegion？

Reply:
Yes.

Reply:
how about decrease some same code？